src/Engines/BaseEngine.cs

```diff
@@ -289,7 +289,24 @@ public void RegisterDisplayEncoder(IResultEncoder encoder)
         ///      into JSON.
         /// </param>
         public void RegisterJsonEncoder(params JsonConverter[] converters) =>
-            RegisterDisplayEncoder(new JsonResultEncoder(this.Logger, converters));
+            RegisterJsonEncoder(null, converters);
+
+        /// <summary>
+        ///      Adds a new result encoder that serializes its output to JSON.
+        ///      Serialization failures are logged, but are not written out
+        ///      as results or displayed.
+        /// </summary>
+        /// <param name="mimeType">
+        ///      MIME type string to be used for this encoded output in the
+        ///      execution result. If null, a default of "application/json"
+        ///      will be used.
+        /// </param>
+        /// <param name="converters">
+        ///      Additional JSON converters to be used when serializing results
+        ///      into JSON.
+        /// </param>
+        public void RegisterJsonEncoder(string mimeType, params JsonConverter[] converters) =>
```

Comment:
I love that this opens the possibility of registering different JSON encoders for different MIME types...!